crates/ra_ide/src/goto_definition.rs

```diff
@@ -39,7 +39,7 @@ pub(crate) fn goto_definition(
                 reference_definition(&sema, &name_ref).to_vec()
             },
             ast::Name(name) => {
-                let def = classify_name(&sema, &name)?.definition();
+                let def = classify_name(&sema, &name)?.into_definition()?;
```

Comment:
I wonder if, in this case, it would be clearer to write the match explicitely, to make it clear which flavor of definition do we pick in the ambigious case.